test/connectors/BusinessDetailsConnectorSpec.scala

```diff
@@ -19,63 +19,93 @@ package connectors
 import _root_.utils.TestConstants._
 import connectors.mocks.TestBusinessDetailsConnector
 import models.ErrorModel
+import models.monitoring.getBusinessDetails.BusinessDetailsAuditModel
 import models.registration._
-import org.scalatest.concurrent.ScalaFutures.convertScalaFuture
+import org.scalatest.BeforeAndAfterEach
 import play.api.http.Status._
-import play.api.libs.json.JsValue
 import play.api.mvc.Request
 import play.api.test.FakeRequest
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+import services.mocks.monitoring.MockAuditService
 import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging.{eventTypeBadRequest, eventTypeInternalServerError, eventTypeServerUnavailable, eventTypeUnexpectedError}
 
-class BusinessDetailsConnectorSpec extends TestBusinessDetailsConnector {
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class BusinessDetailsConnectorSpec extends TestBusinessDetailsConnector with MockAuditService with BeforeAndAfterEach {
 
   implicit val hc = HeaderCarrier()
   implicit val request: Request[_] = FakeRequest()
 
-  "BusinessDetailsConnector.getBusinessDetails" should {
-
+  "BusinessDetailsConnector.getBusinessDetailsUrl" should {
     "GET to the correct url" in {
       TestBusinessDetailsConnector.getBusinessDetailsUrl(testNino) should endWith(s"/registration/business-details/nino/$testNino")
     }
+  }
 
-    def result = TestBusinessDetailsConnector.getBusinessDetails(testNino).futureValue
+  "BusinessDetailsConnector.getBusinessDetails" should {
+    def result(f: GetBusinessDetailsUtil.Response => Any): Any = await(TestBusinessDetailsConnector.getBusinessDetails(testNino).map(v => f(v)))
 
     "parse and return the success response correctly" in {
       mockBusinessDetails(getBusinessDetailsSuccess)
-      result shouldBe Right(GetBusinessDetailsSuccessResponseModel(testMtditId))
+      result { r =>
```

Comment:
I prefer this structure. It makes it clearer that we are actually dealing with a future.